server/models/Order.js

```diff
@@ -446,6 +447,10 @@ function defineModel() {
     });
   };
 
+  Order.cancelActiveOrdersByTierId = function (tierId) {
+    return Order.update({ status: OrderStatuses.CANCELLED }, { where: { TierId: tierId } });
```

Comment:
**[:no_entry_sign: **Blocker**]** We should better filter the orders that we want to update, otherwise we're going to lose a piece of important information for some of them (e.g. "Expired", "Paid", etc).

```suggestion
    return Order.update(
      { status: OrderStatuses.CANCELLED },
      {
        where: {
          TierId: tierId,
          SubscriptionId: { [Op.not]: null },
          status: { [Op.not]: [OrderStatuses.PAID, OrderStatuses.CANCELLED, OrderStatuses.REJECTED, OrderStatuses.EXPIRED] }
        }
    });
```

Reply:
Ops, missed that. I thought about filtering by OrderStatuses.ACTIVE only:

`where: {TierId: tierId, SubscriptionId: { [Op.not]: null }, status: OrderStatuses.ACTIVE }`

wdyt?

Reply:
I wondered about this too when I posted my feedback; we do want to cover at least the `NEW` status on top of `ACTIVE`. When PayPal subscriptions orders are created, the flow is basically
1. Create the order with the "New" status
2. Wait for the PayPal webhook "payment captured" event
3. Record transactions & mark the order as active 

This "payment captured" event might take a few minutes to arrive, and though it's an edge case I think it's the right call to include these in the list of orders to cancel because, on PayPal side, they're already active at this stage.

Because of the way PayPal works, `REQUIRE_CLIENT_CONFIRMATION` is not supposed to happen but I think it's good if we have it too: an order requiring a new 3DSecure confirmation is still kind of active.

As for the other statuses:
- `PLEDGED` is not supposed to happen, but I don't think it hurts to have it
- `PENDING` is not supposed to happen either, and https://github.com/opencollective/opencollective/issues/2734 will probably be implemented with Stripe.
- I suppose `ERROR` could happen on existing recurring contributions if the contributor doesn't have enough balance on their PayPal account and then later add a credit card to it

That's why I suggested having an opt-out filter, but feel free to use an opt-in approach if it makes more sense.